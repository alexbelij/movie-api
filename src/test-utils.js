/* @flow */

import R from 'ramda';

const childTypesForArray = R.memoize(
  R.pipe(R.map(R.type), R.uniq),
);

const modelFromObject = (obj: ?Object) => R.map(
  R.cond([
    [
      R.allPass([
        R.pipe(R.type, R.equals('Array')),
        R.pipe(childTypesForArray, R.length, R.equals(1)),
      ]),
      (val: Array<any>) => `Array<${R.pipe(childTypesForArray, R.head)(val)}>`,
    ],
    [R.pipe(R.type, R.contains(R.__, ['Object', 'Array'])), modelFromObject],
    [R.T, R.type],
  ]),
  obj,
);

const modelFromFirstElement = R.pipe(
  R.head,
  modelFromObject,
);
const modelFromFirstElementAtPath = (path: string) => R.pipe(
  R.pathOr([], path.split('.')),
  modelFromFirstElement,
);

export { modelFromObject, modelFromFirstElement, modelFromFirstElementAtPath };
