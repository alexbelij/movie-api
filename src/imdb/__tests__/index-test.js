/* @flow */

import { modelFromObject } from '../../test-utils';
import Imdb from '../index';

const movieId = 'tt2488496'; // Star Wars: The Force Awakens

describe('IMDB', () => {
  let imdb: Imdb;

  beforeAll(() => {
    imdb = new Imdb();
  });

  it('fetches movie rating by id', async () => {
    expect(modelFromObject(await imdb.getRating(movieId))).toMatchSnapshot();
  });
});
