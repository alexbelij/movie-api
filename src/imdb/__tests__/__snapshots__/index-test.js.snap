exports[`IMDB fetches movie rating by id 1`] = `
Object {
  "imdbRating": "number",
  "imdbRatingVoteCount": "number",
}
`;
